app_spec/test/test.js

```diff
@@ -116,39 +116,172 @@ scenario2.runTape('delete_entry_post', async (t, { alice, bob }) => {
   const content = "Hello Holo world 321"
   const in_reply_to = null
   const params = { content, in_reply_to }
+
+  //commit create_post
   const createResult = await alice.callSync("blog", "create_post", params)
 
   t.ok(createResult.Ok)
 
+
+  //delete entry post
   const deletionParams = { post_address: createResult.Ok }
   const deletionResult = await alice.callSync("blog", "delete_entry_post", deletionParams)
 
   t.notOk(deletionResult.Ok)
 
-  const paramsGet = { post_address: createResult.Ok }
-  const result = bob.call("blog", "get_post", paramsGet)
 
-  t.notOk(result.Ok)
+  //delete should fail
+  const failedDelete = await alice.callSync("blog", "delete_entry_post", { post_address: createResult.Ok })
+  console.log("failed delete " + failedDelete);
+  t.deepEqual(failedDelete.Err,{ Internal: 'Unspecified' });
+
+  //get initial entry
+  const GetInitialParamsResult = alice.call("blog", "get_initial_post", { post_address: createResult.Ok })
+  t.deepEqual(JSON.parse(GetInitialParamsResult.Ok.App[1]),{content: "Hello Holo world 321", date_created: "now" });
+  
+  const entryWithOptionsGet = { post_address: createResult.Ok}
+  const entryWithOptionsGetResult = alice.call("blog", "get_post_with_options", entryWithOptionsGet);
+  t.deepEqual(JSON.parse(entryWithOptionsGetResult.Ok.result.All.items[0].entry.App[1]),{content: "Hello Holo world 321", date_created: "now" })
 })
 
 scenario2.runTape('update_post', async (t, { alice, bob }) => {
   const content = "Hello Holo world 123"
   const in_reply_to = null
   const params = { content, in_reply_to }
+
+  //commit version 1
   const createResult = await alice.callSync("blog", "create_post", params)
+  t.ok(createResult.Ok)
+   //get v1
+  const updatedPostV1 = alice.call("blog", "get_post", { post_address: createResult.Ok })
+  const UpdatePostV1Content = { content: "Hello Holo world 123", date_created: "now" };
+  t.ok(updatedPostV1.Ok)
+  t.deepEqual(JSON.parse(updatedPostV1.Ok.App[1]),UpdatePostV1Content)
+
+  //update to version 2
+  const updatePostContentV2 = { content: "Hello Holo V2", date_created: "now" };
+  const updateParamsV2 = { post_address: createResult.Ok, new_content: "Hello Holo V2" }
+  const UpdateResultV2 = await bob.callSync("blog", "update_post", updateParamsV2)
+  t.ok(UpdateResultV2.Ok)
+
+  //get v2 using initial adderss
+  const updatedPostv2Initial = alice.call("blog", "get_post", { post_address: createResult.Ok })
+  t.ok(updatedPostv2Initial.Ok)
+  t.deepEqual(JSON.parse(updatedPostv2Initial.Ok.App[1]), updatePostContentV2)
+
+  //get v2 latest address
+  const updatedPostv2Latest = alice.call("blog", "get_post", { post_address: UpdateResultV2.Ok })
+  t.ok(updatedPostv2Latest.Ok)
+  t.deepEqual(JSON.parse(updatedPostv2Latest.Ok.App[1]), updatePostContentV2)
+
+
+   //get v2 using initial adderss
```

Comment:
This getting V1, right?
```suggestion
   //get v1 using initial address
```

Reply:
Yes, thank you for that!